File: src/Projects/Android/PaymentAppWithDB2/app/src/main/java/org/csystem/android/app/paymentapp/viewmodel/LoginActivityListenerViewModel.java
package org.csystem.android.app.paymentapp.viewmodel;

import org.csystem.android.app.paymentapp.LoginActivity;

import java.lang.ref.WeakReference;

public class LoginActivityListenerViewModel {
    private final WeakReference<LoginActivity> m_weakReference;

    public LoginActivityListenerViewModel(LoginActivity activity) {
        m_weakReference = new WeakReference<>(activity);
    }

    public void handleLoginButton()
    {
        m_weakReference.get().loginButtonClicked();
    }

    public void handleCloseButton()
    {
        m_weakReference.get().finish();
    }
}

File: src/Libraries/CSDLibraries/org-csystem-android-activity/src/main/java/org/csystem/android/activity/viewmodel/LoginActivityListenerViewModel.kt
// LoginActivityListenerViewModel
package org.csystem.android.activity.viewmodel

import org.csystem.android.activity.LoginActivity
import java.lang.ref.WeakReference

internal class LoginActivityListenerViewModel(activity:LoginActivity)
{
    private val mWeakReference = WeakReference(activity)

    fun handleLoginButton() = mWeakReference.get()?.loginButtonClicked()
    fun handleCloseButton() = mWeakReference.get()?.loginButtonClicked()
}